src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBinaryWriter.cs

```diff
@@ -1218,280 +1149,212 @@ private static void CheckArray(Array array, int offset, int count)
                 throw System.Runtime.Serialization.DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException(nameof(count), SR.Format(SR.SizeExceedsRemainingBufferSpace, array.Length - offset)));
         }
 
-        public override unsafe void WriteArray(string? prefix, string localName, string? namespaceUri, bool[] array, int offset, int count)
+        public override void WriteArray(string? prefix, string localName, string? namespaceUri, bool[] array, int offset, int count)
         {
+            CheckArray(array, offset, count);
+            if (count > 0)
             {
-                CheckArray(array, offset, count);
-                if (count > 0)
-                {
-                    fixed (bool* items = &array[offset])
-                    {
-                        UnsafeWriteArray(prefix, localName, namespaceUri, XmlBinaryNodeType.BoolTextWithEndElement, count, (byte*)items, (byte*)&items[count]);
-                    }
-                }
+                var bytes = MemoryMarshal.Cast<bool, byte>(array.AsSpan(offset, count));
```

Comment:
bools aren't necessarily stored as 0 and 1.  Is this going to always do the same thing it did before?